src/app.js

```diff
@@ -0,0 +1,44 @@
+let localStorage = [];
+
+if (typeof window !== 'undefined' && window.localStorage) {
+  localStorage = window.localStorage;
+}
+
+const updateLocalStorage = (tasks) => {
+  tasks.forEach((task, index) => {
+    task.index = index + 1;
+  });
+
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const addTask = (tasks, description) => {
+  const newTask = {
+    description,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  tasks.push(newTask);
+  updateLocalStorage(tasks);
+};
+const deleteTask = (tasks, taskId) => {
+  if (taskId >= 0 && taskId < tasks.length) {
+    tasks.splice(taskId, 1);
+    for (let i = taskId; i < tasks.length; i += 1) {
+      tasks[i].index = i + 1;
+    }

```

Comment:
- To reduce activity in Loops, as a JS performance best practice, try accessing the length property outside the loop to make the loop run faster.

Reply:
- Refactored `for` loop to conform to JS performance best practices.

Reply:
Yeah, I really appreciated you efforts in following all the best JavaScript Practices.